core/include/bitarithm.h

```diff
@@ -138,6 +138,35 @@ uint8_t bitarithm_bits_set_u32(uint32_t v);
 
 /* implementations */
 
+static inline unsigned bitarithm_msb(unsigned v)
+{
+#if defined(BITARITHM_HAS_CLZ)
+    return 8 * sizeof(v) - __builtin_clz(v) - 1;
+#else
+    register unsigned r; /* result of log2(v) will go here */
+
+#if ARCH_32_BIT
+    register unsigned shift;
+
+    /* begin{code-style-ignore} */
+    r =     (v > 0xFFFF) << 4; v >>= r;
+    shift = (v > 0xFF  ) << 3; v >>= shift; r |= shift;
+    shift = (v > 0xF   ) << 2; v >>= shift; r |= shift;
+    shift = (v > 0x3   ) << 1; v >>= shift; r |= shift;
+                                            r |= (v >> 1);
```

Comment:
Could you move this back to `bitarithm.c` and name it e.g. `bitarith_msb_32bit_no_native_clz()`? Then, this `static inline` function could call that function, if `__builtin_clz()` is only provided as slow software fallback implementation.

Also: I think this code could profit from `IS_ACTIVE()`

Reply:
Yea that's much cleaner.